crates/ruff_linter/resources/test/fixtures/flake8_markupsafe/MS001.py

```diff
@@ -0,0 +1,10 @@
+import flask
+from markupsafe import Markup, escape
+
+content = "<script>alert('Hello, world!')</script>"
+Markup(f"unsafe {content}")  # MS001
+flask.Markup("unsafe {}".format(content))  # MS001
+Markup("safe {}").format(content)
+flask.Markup(b"safe {}", encoding='utf-8').format(content)
+escape(content)
+Markup(content)  # MS001
```

Comment:
Could you please add a couple of more test cases?

Examples:
- f-string formatting
- old-style formatting, `Markup("<em>%s</em> " % ("foo & bar",))
- Explicit keyword arguments, e.g. `Markup(object="hello world")`

https://markupsafe.palletsprojects.com/en/stable/escaping/#markupsafe.Markup

Reply:
Detecting keyword arguments would only reliably work for `markupsafe.Markup` and `flask.Markup` but not for equivalent classes, so I'm not sure it's worth adding extra complexity until type inference can easily give us the first positional argument regardless of how the function was called. And even then it's somewhat questionable, since code searching for `Markup(object=` yields zero results, I don't expect anyone to ever call it this way.

On the other hand, it is a security rule, so we could consider emitting an error if there isn't at least one positional argument.

But I knew I forgot about something, and it was old-style formatting, f-strings do already have a test case (the very first one).

Reply:
> Detecting keyword arguments would only reliably work for markupsafe.Markup and flask.Markup but not for equivalent classes, so I'm not sure it's worth adding extra complexity until type inference can easily give us the first positional argument regardless of how the function was called. And even then it's somewhat questionable, since code searching for Markup(object= yields zero results, I don't expect anyone to ever call it this way.

I was only suggesting to add them as test cases, and agree that we do not have to support. I think it's a good idea to include known syntactially valid negative test cases ahead of time for posterity.

Reply:
Yeah, that's fair. I did include corresponding test cases.

Reply:
> But I knew I forgot about something, and it was old-style formatting, f-strings do already have a test case (the very first one).

Overlooked the f-string. For completeness we can consider other valid but less frequent cases, e.g. multi-line strings, implicit string concatenation,  with and without comments.